amundsen_common/models/lineage.py

```diff
@@ -0,0 +1,37 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, List
+
+from amundsen_common.models.table import Badge
+
+import attr
+from marshmallow_annotations.ext.attrs import AttrsSchema
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class LineageItem:
+    key: str  # down/upstream table/col/task key
+    level: int  # upstream/downstream distance from current resource
+    source: str  # database this resource is from
+    badges: Optional[List[Badge]] = None
+    usage: Optional[int] = None  # statistic to sort lineage items by
+
+
+class LineageItemSchema(AttrsSchema):
+    target = LineageItem
+    register_as_scheme = True
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class Lineage:
+    key: str  # current table/col/task key
+    direction: str  # upstream/downstream/both
```

Comment:
any reason we need direction and depth if we have `lineage_entities_upstream` and lineage_entities_downstream?

Reply:
LineageItem seems to include level already?

Reply:
Just to persist the selection (upstream, downstream, both) send over in the request. Similarly depth might be needed because it tells you how many levels were requested for this response.